components/server/src/workspace/workspace-starter.ts

```diff
@@ -686,9 +677,9 @@ export class WorkspaceStarter {
         return undefined;
     }
 
-    protected async getAdditionalImageAuth(projectEnvVars: ProjectEnvVar[]): Promise<Map<string, string>> {
+    protected async getAdditionalImageAuth(envVars: ResolvedEnvVars): Promise<Map<string, string>> {
         const res = new Map<string, string>();
-        const imageAuth = projectEnvVars.find((e) => e.name === "GITPOD_IMAGE_AUTH");
+        const imageAuth = envVars.project.find((e) => e.name === "GITPOD_IMAGE_AUTH");
```

Comment:
@svenefftinge this is the major reason why we need to return all, although we could change that this method itself fetches a required env var.

Reply:
I wonder why this env var needs to be set on project-level and not anyhow.

Reply:
@svenefftinge I removed other layers, but kept this one for now. I think someone else can address it later if needed. Looking at the original PR and docs it seems to be that intent was to allow it only on project env vars.